Lib/http/server.py

```diff
@@ -998,6 +1008,7 @@ def test(HandlerClass=BaseHTTPRequestHandler,
         except KeyboardInterrupt:
             print("\nKeyboard interrupt received, exiting.")
             sys.exit(0)
+    return server
```

Comment:
I don't like this. The test function only "serves" the server and exits upon Ctrl+C. It's not meant to return the server instance. Instead, I suggest we have a function that constructs the server and `test` would become:

```py
with _make_server(...) as httpd:
    print(...)
    try:
        httpd.serve_forever()
     except ...
``

In the tests you can directly use `_make_server`.